### PR TITLE
Bugfix/prod hotfixes 20200318

### PR DESCRIPTION
**Pull request by j5awry**
Added prod hotfix, skipping unfound branches/repos. Parameterized parallel builds, allowings users to set number of jobs (with a default of 'as many as possible.'). Add in ways to install extras like tox and launchpadlib. Expand README with more information

### CODE REVIEW

**Comment by rcj4747** [comment]
You have a commit in the PR just called 'test' that adds a newline.  Please drop that commit.

**Review by rcj4747 — COMMENTED** [review]
The title and description of the 'Bring in hotfixes' commit doesn't add value as-is and needs to be changed.

1. Don't mention 'CPC' on an external PR because the acronym doesn't have meaning outside the company (it hardly has meaning inside the company) ;)
2. These were 'hotfixes' but that doesn't tell us what they address/change just that someone was running with patches on the upstream code for some time.
3. Consider organizing the changes functionally by patch.

**Review by rcj4747 — CHANGES_REQUESTED** [review]
The 'Add prod hotfixes, parameterize parallel' commit message should be renamed or reorganized.  "Add prod hotfixes" does not inform the reader as to what the changes do or why.

File: README.md
Comment (rcj4747) [inline README.md:37]: Please describe the function of:

* 'owners' vs 'branches
* max-age
* review-count

File: README.md
Comment (rcj4747) [inline README.md]: 'owners' was added 3 years ago (prior to v0.2) in commit 6edf6350.  What is the rationale for making this experimental? Was there some discussion of this policy?  I know that we've been using this in production for some time now.  Because I know that we've been using the feature forever the 2nd sentence reads as "if it's broken why don't you fix my code" and it rubs me the wrong way.

File: README.md
Comment (rcj4747) [inline README.md]: Is "all the launchpad capabilities" or is "any launchpad capability" correct?  If it's any then I'd like to see launchpadlib by default as Canonical's code hosting solution is launchpad and it should work out of the box.  If it is only a subset of capabilities then the readme should inform the user what they are/aren't so they can choose.

Comment (rcj4747) [reply README.md]: Thinking about this more, is 'tox' and dependencies large enough to push off in 'extras' rather than just making it a normal requirement?  pip installing tox in a new virtualenv pulled in: toml, py, filelock, six, appdirs, distlib, virtualenv, pluggy, pyparsing, packaging, tox which totaled about 5mb

**Comment by j5awry** [comment]
rebased and pushed better commit messages.

File: setup.py
Comment (rcj4747) [inline setup.py]: I don't see deepcopy used in this commit.

File: README.md
Comment (j5awry) [reply README.md]: I think this might be a question for @fginther ? I wasn't sure if not adding tox and launchpadlib were oversights or if they were separated for a reason. I went with "separated for a reason." I actually agree that they should just get pulled in normally, but I'll defer to longer maintainers.

File: README.md
Comment (j5awry) [reply README.md]: Had IRC conversation with @rcj4747 . Discussed the issues around the `owner` stanza and the best way of communicating the issue. At present `get_branches_for_owner` isn't working as expected. I'll be changing this to be clear that we are aware the capabilities aren't as we'd like, that it requires  more testing and development, and will add TODO items to the README

File: README.md
Comment (rcj4747) [reply README.md:37]: @j5awry I would still like to see the field documented.

File: README.md
Comment (rcj4747) [reply README.md]: We chatted this morning and decided to include tox by default and a commit has been pushed for that.

File: setup.py
Comment (rcj4747) [reply setup.py]: Fixed in latest commit.

File: README.md
Comment (rcj4747) [inline README.md]: Is this 'reviews' like reviewers on MPs or MPs themselves?

**Review by rcj4747 — CHANGES_REQUESTED** [review]
I've posted a question about the 'review-count' config option

File: README.md
Comment (fginther) [reply README.md]: FYI, this is meant to specify the minimum number of review boxes to display for an MP, implying a minimum number of reviewers.

**Review by fginther — COMMENTED** [review]
Thanks for the much improved README.

**Review by rcj4747 — APPROVED** [review]
+1

**Comment by philroche** [comment]
+1 Thanks

**Review by philroche — APPROVED** [review]

**Pull request merged**
